### ISSUE TITLE
[Feature]: Will vLLM support flash-attention 3 ?

### ISSUE THREAD

**Issue by jorgeantonio21** [vllm-project/vllm#11372]
### 🚀 The feature, motivation and pitch

Flash attention 3 seems very promising for running efficient LLM inference on NVIDIA Hopper cards. Are there any plans to support it in the future ?

https://github.com/Dao-AILab/flash-attention

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by DarkLight1337** [comment]
@WoosukKwon can you help answer this?

**Comment by DarkLight1337** [comment]
Also see #11194

**Comment by jorgeantonio21** [comment]
Really appreciate the effort guys ! Any estimate when this will be available ?

**Comment by mgoin** [comment]
Once the FA3 refactor branch lands (I believe currently at https://github.com/Dao-AILab/flash-attention/tree/decode), it should be ready to integrate. Hopefully within a few weeks

**Comment by jorgeantonio21** [comment]
@mgoin I remember that integrating FA2 with vLLM introduced a bunch of issues with block sizes less than 256 and how . I presume the same issues will have to be addressed with FA3..

**Comment by mgoin** [comment]
I think we will support small block sizes from the start of FA3 support, this is part of the reason why it is taking longer as we have this as a requirement.

**Comment by jorgeantonio21** [comment]
@mgoin @DarkLight1337 is there any progress on this ?

**Comment by mgoin** [comment]
We have landed initial support for FA3 with this PR! https://github.com/vllm-project/vllm/pull/12093

Currently only Hopper support is included due to binary size concerns, but look forward to it in this upcoming release.

**Comment by mgoin** [comment]
Tracking remaining features here https://github.com/vllm-project/vllm/issues/12429

**Comment by jorgeantonio21** [comment]
This is great work team, vLLM works much faster for me now. Thank you !
